src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -156,6 +160,7 @@ export class Kernel implements IKernel {
             try {
                 await this.notebook.restartKernel(this.launchTimeout);
                 await this.initializeAfterStart();
+                await addSysInfo(SysInfoReason.Restart, notebookDocument, this.notebook);
```

Comment:
It didn't seem like it needed to be a class method (even a private one), as it doesn't access any instance variables.

It's also not in `initializeAfterStart` because the method that calls `initializeAfterStart` is what has context on the `SysInfoReason`, which determines the message that gets displayed. Otherwise we need to pass the `reason` parameter through, or infer the reason from the presence of `this.restartingPromise` or a similar flag, both of which seem less precise than explicitly passing the reason as an argument.